CppCustomVisualizer2/dll/ChildVisualizer.cpp

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#include "stdafx.h"
+#include "ChildVisualizer.h"
+
+HRESULT CChildVisualizer::Initialize(
+    _In_ DkmVisualizedExpression* pVisualizedExpression,
+    _In_ size_t vectorSize,
+    _In_ size_t parentIndex
+)
+{
+    m_pVisualizedExpression = pVisualizedExpression;
+    m_vectorSize = vectorSize;
+    m_parentIndex = parentIndex;
+    return S_OK;
+}
+
+HRESULT CChildVisualizer::CreateEvaluationResult(
+    _In_ DkmString* pName,
+    _In_ DkmString* pFullName,
+    _In_ DkmString* pType,
+    _In_ DkmRootVisualizedExpressionFlags_t flags,
+    _In_opt_ DkmVisualizedExpression* pParent,
+    _In_ DkmInspectionContext* pInspectionContext,
+    _In_ size_t index,
+    _Deref_out_ DkmEvaluationResult** ppResultObject
+)
+{
+    HRESULT hr = S_OK;
+
+    CComPtr<DkmPointerValueHome> pPointerValueHome = DkmPointerValueHome::TryCast(m_pVisualizedExpression->ValueHome());
+    if (pPointerValueHome == nullptr)
+    {
+        // This sample only handles visualizing in-memory Sample structures
+        return E_NOTIMPL;
+    }
+
+    // Create method for DkmDataAddress takes a runtime instance.
+    CComPtr<DkmDataAddress> pAddress;
+    hr = DkmDataAddress::Create(m_pVisualizedExpression->InspectionContext()->RuntimeInstance(), pPointerValueHome->Address(), NULL, &pAddress);
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+
+    CString strValue;
+    strValue.Format(L"%i", index);
```

Comment:
>index [](http://example.com/codeflow?start=27&length=5)

cast to int to make sure there is always agreement between the printf argument size and the value.

Reply:
Seems like the correct way to print index is `%zu`. I'll change this.